deps-mvn-indexing/src/main/java/com/connexta/osgeyes/index/JarPackagesIndexCreator.java

```diff
@@ -0,0 +1,129 @@
+package com.connexta.osgeyes.index;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.Field.Index;
+import org.apache.maven.index.ArtifactContext;
+import org.apache.maven.index.ArtifactInfo;
+import org.apache.maven.index.IndexerField;
+import org.apache.maven.index.IndexerFieldVersion;
+import org.apache.maven.index.context.IndexCreator;
+import org.apache.maven.index.util.zip.ZipFacade;
+import org.apache.maven.index.util.zip.ZipHandle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Makes all packages contained in a jar indexed and searchable. */
+@Singleton
+@Named(JarPackagesIndexCreator.ID)
+public class JarPackagesIndexCreator implements IndexCreator {
+
+  // Useful to reference within package scope
+  static final String ID = "deps/jar-packages";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JarPackagesIndexCreator.class);
+
+  private static final IndexerField JAR_PACKAGES_FIELD =
+      new IndexerField(
+          MvnOntology.JAR_PACKAGES,
+          IndexerFieldVersion.V3,
+          MvnOntology.JAR_PACKAGES.getFieldName(),
+          MvnOntology.JAR_PACKAGES.getDescription(),
+          Field.Store.YES,
+          Index.ANALYZED);
+
+  private static final List<IndexerField> FIELDS = Collections.singletonList(JAR_PACKAGES_FIELD);
+
+  private static final List<String> DEPS = Collections.singletonList("min");
+
+  @Override
+  public String getId() {
+    return ID;
+  }
+
+  @Override
+  public List<String> getCreatorDependencies() {
+    return DEPS;
+  }
+
+  @Override
+  public Collection<IndexerField> getIndexerFields() {
+    return FIELDS;
+  }
+
+  @Override
+  public void populateArtifactInfo(ArtifactContext artifactContext) throws IOException {
+    final ArtifactInfo info = artifactContext.getArtifactInfo();
+    final Map<String, String> attributes = info.getAttributes();
+    final File artifactFile = artifactContext.getArtifact();
+    if (artifactFile != null && artifactFile.isFile() && artifactFile.getName().endsWith(".jar")) {
+      ZipHandle handle = null;
+      try {
+        handle = ZipFacade.getZipHandle(artifactFile);
+        List<String> paths =
+            handle.getEntries().stream()
+                .filter(s -> s.endsWith(".class"))
+                .map(
+                    s -> {
+                      String[] split = s.split("/");
+                      return String.join(".", Arrays.copyOf(split, split.length - 1));
+                    })
```

Comment:
❓ Is it possible to rewrite this as two separate map operations? 
```
...
.filter(s -> s.endsWith(".class"))
.map(s -> s.split("/"))
.map(sa -> String.join(".", Arrays.copyOf(sa, sa.length - 1)))
.distinct()
...
```

Reply:
No, because `.map(s -> s.split("/"))` is a `Stream<String[]>`, instead of `Stream<String>`. The split and join need to be done atomically on each String in the stream.

Reply:
Okay that's fine. Then let's use a private static function for it. 
```
...
// This will probably be collapsible to a method ref
.map(s -> classFileToPackageNotation(s))
...

private static String classFileToPackageNotation(String classFile) {
  String[] split = classFile.split("/");
  return String.join(".", Arrays.copyOf(split, split.length - 1));
}
```

Reply:
Done.